connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -590,3 +610,114 @@ function sortForIncrementalUpdate(changedList: DriveItem[]) {
 
   return sortedDriveItemList;
 }
+
+async function getDeltaData({
+  client,
+  node,
+}: {
+  client: Client;
+  node: MicrosoftNodeResource;
+}) {
+  if (!node.deltaLink) {
+    throw new Error(
+      `Delta link not found for root node resource ${JSON.stringify(node.toJSON())}`
+    );
+  }
+
+  try {
+    return await getFullDeltaResults(client, node.internalId, node.deltaLink);
+  } catch (e) {
+    if (e instanceof GraphError && e.statusCode === 410) {
+      // API is answering 'resync required'
+      // we repopulate the delta from scratch
+      return await getFullDeltaResults(client, node.internalId);
+    }
+    throw e;
+  }
+}
+
+async function isFolderMovedInSameRoot({
+  connectorId,
+  folder,
+  internalId,
+}: {
+  connectorId: ModelId;
+  folder: DriveItem;
+  internalId: string;
+}) {
+  if (!folder.parentReference) {
+    throw new Error(`Unexpected: parent reference missing: ${folder}`);
+  }
+
+  const oldResource = await MicrosoftNodeResource.fetchByInternalId(
+    connectorId,
+    internalId
+  );
+
+  if (!oldResource) {
+    // the folder was not moved internally since we don't have it
+    return false;
+  }
+
+  const oldParentId = oldResource.parentInternalId;
+
+  if (!oldParentId) {
+    // this means it is a root
+    return false;
+  }
+
+  const newParentId = getParentReferenceInternalId(folder.parentReference);
+
+  return oldParentId !== newParentId;
+}
+
+async function updateDescendantsParentsInQdrant({
+  folder,
+  dataSourceConfig,
+  startSyncTs,
+}: {
+  folder: MicrosoftNodeResource;
+  dataSourceConfig: DataSourceConfig;
+  startSyncTs: number;
+}) {
+  const children = await folder.fetchChildren();
```

Comment:
`nit`: I added an optional argument where you can pass a list of `nodeTypes` to filter on. Not gonna change much but you could do:
```typescript
const children = await folder.fetchChildren(["file", "folder"]);
```

Reply:
In that case, i'd rather fetch all children and have the code err if they're not files or folders --- since it should not happen (not sure i'm clear :sweat_smile: happy to clarify irl)